packages/core/src/Coordinator.js

```diff
@@ -244,6 +246,17 @@ export class Coordinator {
 
     // connect filter selection
     connectSelection(this, client.filterBy, client);
+
+    // Bookkeep all clause sources
+    if (isClauseSource(client)) {
```

Comment:
We need a more general solution. The technique of inspecting the client based on known types (such as Plot) is breaking our intended abstraction barriers. Clients and clause sources are often paired, but in general can be distinct. I think we need to find a way to have interactors register/deregister themselves; this will require that they have access to the corresponding coordinator instance, which may involve refactoring parts of the other packages.

Reply:
Should be resolved now, let me know what you think of the new way I'm doing it.